src/lib/spinel/radio_spinel_impl.hpp

```diff
@@ -253,7 +255,11 @@ void RadioSpinel<InterfaceType, ProcessContextType>::Init(bool aResetRadio, bool
         DieNow(exitCode);
     }
 
-    SuccessOrDie(CheckRcpApiVersion(supportsRcpApiVersion));
+    if (!aSkipRcpApiVersionCheck)
+    {
+        SuccessOrDie(CheckRcpApiVersion(supportsRcpApiVersion));
+    }
+
     SuccessOrDie(CheckRadioCapabilities());
```

Comment:
We also check RCP capabilities (via CheckRadioCapabilities). Should we also provide arguments to bypass it for cases like firmware update?

Reply:
I was initially considering to include the radio cap check, but decided not to do this, to keep the change simple. I was concerned that not doing `CheckRadioCapabilities` would keep `mRadioCaps` as empty or zero and if it may have side-effects (it would impact not just `RadioSpinel` but OT core and sub-mac). 

Looking now at all the places `mRadioCaps` is used I think it should be fine, so will change it.

Reply:
Changed in new pushed commit.